src/main/webapp/script.js

```diff
@@ -23,24 +23,28 @@ google.setOnLoadCallback(getChart)
 
 async function getYouTubeComments() { 
   const urlInput = document.getElementById('link-input');
-  const url = cleanseUrl(urlInput.value);
+  const url = extractYouTubeUrl(urlInput.value);
   const response = await fetch("/YouTubeComments?url="+url);
   const comments = await response.json();
   return comments;
 }
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
+ * Extracts video ID from the full URL
+ * Method derived from: 
+ * https://stackoverflow.com/questions/28735459/how-to-validate-youtube-url-in-client-side-in-text-box
+ */
+function extractYouTubeUrl(url) {
+  var videoId = "";
+  var idLength = 11; // Length of video Id's in YouTube videos
+  if (url != undefined || url != '') {
+    var regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=|\?v=)([^#\&\?]*).*/;
```

Comment:
In addition to Cis's comment about testing, I think we should figure out specifically which youtube urls work - it might be most of them, but we should definitely test before deploying

Reply:
Agreed